eksupgrade/models/eks.py

```diff
@@ -0,0 +1,357 @@
+"""Define the models to be used across the EKS upgrade tool."""
+from __future__ import annotations
+
+import base64
+import datetime
+import logging
+from abc import ABC
+from dataclasses import dataclass, field
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+import boto3
+from kubernetes import client as k8s_client
+from kubernetes import config as k8s_config
+
+from .base import AwsRegionResource
+
+try:
+    from functools import cache
+except ImportError:  # pragma: no cover
+    from functools import lru_cache as cache
```

Comment:
(I don't know this) I would have thought this would have been reversed - try for LRU cache and if not available, use generic default cache method